plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtCopyToSecondaryStorageWrapper.java

```diff
@@ -0,0 +1,91 @@
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.hypervisor.kvm.resource.wrapper;
+
+import com.cloud.agent.api.Answer;
+import com.cloud.hypervisor.kvm.resource.LibvirtComputingResource;
+import com.cloud.hypervisor.kvm.storage.KVMStoragePool;
+import com.cloud.hypervisor.kvm.storage.KVMStoragePoolManager;
+import com.cloud.resource.CommandWrapper;
+import com.cloud.resource.ResourceWrapper;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.cloud.utils.script.Script2;
+
+import org.apache.cloudstack.diagnostics.CopyToSecondaryStorageAnswer;
+import org.apache.cloudstack.diagnostics.CopyToSecondaryStorageCommand;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+@ResourceWrapper(handles = CopyToSecondaryStorageCommand.class)
+public class LibvirtCopyToSecondaryStorageWrapper extends CommandWrapper<CopyToSecondaryStorageCommand, Answer, LibvirtComputingResource> {
+    public static final Logger LOGGER = Logger.getLogger(LibvirtCopyToSecondaryStorageWrapper.class);
+    final static String PERM_KEY = "/root/.ssh/id_rsa.cloud";
+    final static String COMMAND = "/bin/bash";
+    final static String DEFAULT_PORT = "3922";
+    final static String DIAGNOSTICS_DIR = "diagnostics_data";
+
+
+    @Override
+    public Answer execute(CopyToSecondaryStorageCommand command, LibvirtComputingResource libvirtComputingResource) {
+
+        final String diagnosticsFile = command.getFileName();
+        final String vmSshIp = command.getSystemVmIp();
+        final String secondaryStorageUrl = command.getSecondaryStorageUrl();
+
+        KVMStoragePoolManager storagePoolMgr = libvirtComputingResource.getStoragePoolMgr();
+        KVMStoragePool secondaryPool;
+
+        // Copy zip file from system VM instance to mounted secondary storage directory
+        boolean success;
+        try {
+            secondaryPool = storagePoolMgr.getStoragePoolByURI(secondaryStorageUrl);
+            String mountPoint = secondaryPool.getLocalPath();
+
+            if (secondaryPool != null) {
+                LOGGER.info(String.format("Copying %s from %s to secondary store %s", diagnosticsFile, vmSshIp, secondaryStorageUrl));
+
+                // /mnt/SecStorage/uuid/diagnostics_data
```

Comment:
this format() eveluates to .... the comment could be improved to help future coders